File: 0002-add-two-numbers/0002-add-two-numbers.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2) {
   // typedef ListNode* node;
    struct ListNode* x=l1;
    struct ListNode* y=l2;
    struct ListNode* dummy=(struct ListNode*)malloc(sizeof(struct ListNode));
    struct ListNode* temp=dummy;
    int sum;
    int carry=0;
    while(x!=NULL&&y!=NULL){
        temp->next=x;
        temp=x;
        sum=x->val+y->val;
        if(sum<9){
            temp->val=sum+carry;
            carry=0;
        }else{
            sum=sum+carry;
            temp->val=(sum%10);
            carry=sum/10;
        }
        x=x->next;
        y=y->next;
        
    }
    while(x!=NULL){
        temp->next=x;
        temp=x;
        if(temp->val<9){
            temp->val+=carry;
            carry=0;
            break;
        }else{
            sum=temp->val+carry;
            temp->val=sum%10;
            carry=sum/10;
        }
        x=x->next;
    }

    while(y!=NULL){
        temp->next=y;
        temp=y;
        if(temp->val<9){
            temp->val+=carry;
            carry=0;
            break;
        }else{
            sum=temp->val+carry;
            temp->val=sum%10;
            carry=sum/10;
        }
        y=y->next;
    }
    if(carry!=0){
        struct ListNode* z=(struct ListNode*)malloc(sizeof(struct ListNode));
        z->val=carry;
        z->next=NULL;
        temp->next=z;
    }
    
    return dummy->next;
    
}

File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
       ListNode head1 = l1;
       ListNode head2 = l2;

       ListNode dummy = new ListNode(-1,null);
       ListNode temp = dummy;

       int carry = 0;
       int sum = 0;


        while(head1!=null && head2!=null){
            sum = head1.val + head2.val + carry;
            carry = sum/10;
            sum = sum%10;

            head1.val = sum;
            temp.next = head1;
            temp = head1;

            head1 = head1.next;
            head2 = head2.next;
        }

        while(head1!=null && carry!=0){
            sum = head1.val + carry;
            carry = sum/10;
            sum = sum%10;

            head1.val = sum;
            temp.next = head1;
            temp = head1;

            head1 = head1.next;

        }

        while(head2!=null && carry!=0){
            sum = head2.val + carry;
            carry = sum/10;
            sum = sum%10;

            head2.val = sum;
            temp.next = head2;
            temp = head2;
            head2 = head2.next;

        }

        if(carry == 0){
            if(head1 == null)temp.next = head2;
            else temp.next = head1;
        }else{
            ListNode newNode = new ListNode(carry,null);
            temp.next = newNode;
        }

       return dummy.next;
    }
}